File: filesystem/src/grammar/tests/validation_data/js/validator_erroneous_code.js
/**
 * JavaScript validation file with syntax errors to test parser robustness
 */

// Syntax error: missing semicolon in a context where it matters
const x = 5
const y = 10;

// Syntax error: using undefined variable
console.log(undefinedVariable);

// Syntax error: mismatched brackets
function brokenFunction() {
  return {
    name: "test",
    value: 42
  ];
}

// Syntax error: invalid property name
const obj = {
  @invalid: true,
  'valid': true
};

// Syntax error: reserved word as variable
const class = "some value";

// Syntax error: assignment in conditional
if (x = 5) {
  console.log("This will always be true");
}

// Syntax error: using var after declaration
const z = 10;
var z = 20;

// Syntax error: unexpected token
const arr = [1, 2, 3,];,

// Syntax error: double comma in array
const badArray = [1, 2,, 3];

// Syntax error: missing closing quote
const stringError = "Unclosed string;

// Syntax error: invalid escape sequence
const escapeError = "Invalid escape: \z";

// Syntax error: octal literals not allowed in strict mode
"use strict";
const octalError = 0765;

// Syntax error: duplicate parameter names
function duplicateParams(a, a) {
  return a;
}

// Syntax error: unexpected line break in string
const multilineString = "This string has
a line break";

// Syntax error: invalid regular expression
const regex = /[a-z/;

// Syntax error: invalid hex number
const hex = 0xZ123;

// Syntax error: with statement in strict mode
"use strict";
with (Math) {
  console.log(PI);
}

// Syntax error: invalid labeled statement
test: const labelVar = 5;

// Syntax error: invalid use of await outside async function
function notAsync() {
  const data = await fetch('/api/data');
  return data;
}

// Syntax error: illegal break statement
function illegalBreak() {
  if (true) {
    break;
  }
}

// Syntax error: invalid delete
delete x;

// Syntax error: invalid for-loop initialization
for (var i = 0, j = 10; i < j i++) {
  console.log(i);
}

// Syntax error: missing colon in object
const missingColon = {
  prop1 "value",
  prop2: "value2"
};

// Syntax error: function declaration without name
function() {
  return "anonymous";
}

// Syntax error: trailing comma in object literal (in older JS)
const oldStyleObject = {
  prop1: "value1",
  prop2: "value2",
};

// Syntax error: calling with wrong this context
console.log.call();

// Multiple errors in one block
function multipleErrors() {
  const x = {;
  return [1, 2,;
  if true {
    console.log("error");
  }
} 

File: filesystem/src/grammar/tests/validation_data/py/validator_erroneous_code.py
"""
Python validation file with syntax errors to test parser robustness
"""

# Syntax error: mismatched parentheses
def function_with_mismatched_parens():
    return ((5 + 3) * 2))

# Syntax error: invalid indentation
def function_with_bad_indentation():
    if True:
    print("Wrong indentation")
        print("Inconsistent indentation")
          return True

# Syntax error: using a keyword as a variable name
class = "This is a class variable"

# Syntax error: missing colon in if statement
if True
    print("Missing colon")

# Syntax error: using assignment operator in conditional
if x = 5:
    print("Using = instead of ==")

# Mix of tabs and spaces
def mixed_tabs_and_spaces():
	var1 = 10  # Tab
    var2 = 20  # Spaces
	return var1 + var2  # Tab again

# Syntax error: missing commas in list
my_list = [1 2 3 4]

# Syntax error: unclosed string literal
unclosed_string = "This string has no closing quote

# Wrong f-string syntax
f_string_error = f"Value is {value + "

# Syntax error: standalone expression
"This is just a string not assigned to anything"
5 + 3

# Invalid decorator
@not_a_decorator(arg1, arg2)
def decorated_function():
    pass

# Using return outside of function
return "This is outside a function"

# Mixing tabs and spaces in the same line
def mixed_indentation():
    value = 5	  # Mixed tabs and spaces
    return value

# Incorrectly nested blocks
def nested_blocks_error():
    try:
        x = 1 / 0
    except ZeroDivisionError:
        print("Divide by zero")
    finally:
        print("Cleanup")
        finally:  # Double finally
            print("This shouldn't be here")

# Unmatched closing brackets
def unmatched_brackets():
    data = {
        "key1": [
            "value1",
            "value2",
        ],
    }}}  # Extra closing braces

# Using continue outside of loop
continue

# Multiple syntax errors in one function
def multiple_errors(param1,, param2):  # Extra comma
    if param1 == None  # Missing colon
        return None
    elif param1 = param2:  # Assignment instead of comparison
        print("Equal")
    return param1.

# Invalid use of walrus operator
if items := len([1, 2, 3]) > 2  # Missing colon
    print(items)

# Calling method on nothing
.method()

# Invalid augmented assignment
x += * 10 